action_server/src/action_server/actions/example_action.py

```diff
@@ -0,0 +1,133 @@
+from action import Action, ConfigurationData
+from robot_skills.util.entity import Entity
+
+
+class ExampleAction(Action):
+    """
+    The ExampleAction class explains how Actions, as handled by the TaskManager, work.
+    """
+    def __init__(self):
+        # Call the base class constructor
+        Action.__init__(self)
+        """
+        The base class constructor initializes the following fields:
+         - self._config_result = ConfigurationResult()
+         - self._execute_result = ActionResult()
+        Their fields should be assigned meaningful values for the caller to give decent feedback to the user.
+
+        Here, you can specify which robot skills are required to perform this action, you can specify what fields are
+        required in the semantics of the task, and what information from previous tasks is required.
+        """
+
+        """
+        You can set the required robot skills here. During the configure step, the given robot object is checked for
+        their availability.
+        """
+        self._required_skills = ['head', 'hmi', 'base']
+
+        """
+        Similarly, you can set the required keys in the knowledge. Their availability is also checked in the configure
+        step.
+        """
+        self._required_passed_knowledge = []
+
+        """
+        Similarly, you can set the required keys in the semantics. Their availability is also checked in the configure
+        step. However, a missing field means that the client may want to add this information by posing a follow-up
+        question. Therefore, this is a dict mapping from the required fields to the prompts requesting the user for
+        the specific missing information.
+
+        If information is missing, the json path to the missing field is returned to the client, so the client knows
+        which field to add.
+        """
+        self._required_field_prompts = {'object': " What object would you like me to manipulate? "}
+
+        """
+        Objects that should be passed on to a next action should be instantiated here.
+        """
+        self._object = Entity()
```

Comment:
Can you give an example with a real entity, like a coke can?
Or a more advanced example with a reference from the previous action?

Reply:
I think this example wouldn't even work, now that I look at it again. I'll fix something. BTW, I can't give an example with a specific object, because the idea of an action is that the type of object is always a parameter in the semantics of a task.